Mathlib/Computability/NFA.lean

```diff
@@ -104,6 +152,159 @@ def accepts : Language α := {x | ∃ S ∈ M.accept, S ∈ M.eval x}
 theorem mem_accepts {x : List α} : x ∈ M.accepts ↔ ∃ S ∈ M.accept, S ∈ M.evalFrom M.start x := by
   rfl
 
+variable (M) in
+/-- `M.Path` represents a concrete path through the NFA from a start state to an end state
+for a particular word.
+
+Note that due to the non-deterministic nature of the automata, the can be more than one `Path`
+for a given word.
+
+Also note that this is `Type` and not a `Prop`, so that we can speak about the properties
+of a particular `Path`, such as the set of states visited along the way (defined as `Path.supp`). -/
+inductive Path : σ → σ → List α → Type (max u v)
+  | nil (s : σ) : Path s s []
+  | cons (t s u : σ) (a : α) (x : List α) :
+      t ∈ M.step s a → Path t u x → Path s u (a :: x)
+
+/-- Knowing that a path must exist, construct one particular example. -/
+noncomputable def path_of_evalFrom {s t : σ} {x : List α} (h : t ∈ M.evalFrom {s} x) :
+    M.Path s t x :=
```

Comment:
There are a few things that are wrong with this definition:
1. Its name is incorrectly cased
2. The relevant Type-valued arguments are implicit, meaning that it will unhelpfully print as `path_of_evalFrom ⋯` in the infoview (proofs are elided)
3. The path doesn't depend on the proof, and therefore I wonder whether this definition is useful at all. I think you instead want a theorem stating `t ∈ M.evalFrom {s} x ↔ Nonempty (M.Path s t x)`